### ISSUE TITLE
Bug - Polity page maps view sometimes missing shapes on "full polity" mode

### ISSUE THREAD

**Issue by Zsajk** [Seshat-Global-History-Databank/seshat#144]
### Description of Improvement

In some cases, if a polity is selected which is a component from another polity, on the polity page this polity is not visible anymore if selected as full polity. 

example year 1390 here : https://seshat-db.com/core/polity/568

### Dependencies

#163 

### Definition of Done

- [ ] The feature has been developed on a feature branch.
- [ ] The feature branch has been tested in the Staging website.
- [ ] A pull request has been created for the feature branch to be merged into the dev branch.

**Comment by edwardchalstrey1** [comment]
Looking into this now, I think it's a separate issue to #163 and is specific to polities that are part of a personal union at some point. I will make this a priority to fix in the new year if I don't have time before then. Having said that, it may not be much of a problem after the polities/component switch is removed

**Comment by Zsajk** [comment]
Its maybe more complicated than that.

For example we have composites of polities which have a different seshat ids , while the component has another id. And to add to that, the component with its own id has a personal union with another polity outside of that composite .


For example in this case : 

{ "type": "Feature", "properties": { "Name": "(Kingdom of Bohemia)", "FromYear": 1305, "ToYear": 1313, "Area": 155832.20202684071, "Type": "POLITY", "Wikipedia": "Kingdom of Bohemia", "SeshatID": "cz_bohemian_k_1", "MemberOf": "(Holy Roman Empire);(Personal union of Kingdom of Bohemia with Kingdom of Poland)", "Components": "Kingdom of Bohemia" }

So here Kingdom of Bohemia has its own id cz_bohemian_k_1 at the same time is member of the Holy Roman Empire 

HRE has its own id: de_empire_3 and several components :


 { "type": "Feature", "properties": { "Name": "(Holy Roman Empire)", "FromYear": 1305, "ToYear": 1313, "Area": 1054365.058456905, "Type": "POLITY", "Wikipedia": "Holy Roman Empire", "SeshatID": "de_empire_3", "MemberOf": "", "Components": "Republic of Florence;Patriarchate of Aquileia;County of Brabant;(Kingdom of Bohemia);House of Wittelsbach;Swiss Confederation;County of Holland;Teutonic Order;House of Luxembourg;Principality of Orange;House of Ascania;Kingdom of Bohemia;House of Habsburg;Margraviate of Brandenburg;County of Savoy;Duchy of Bavaria;Holy Roman Empire Minor States" }, 


At the same time the personal union in the previous example is with the Kingdom of Poland which is not part of the HRE but unified with only Bohemia for some years because the king ruled both in a personal union which is represented here: 

{ "type": "Feature", "properties": { "Name": "(Personal union of Kingdom of Bohemia with Kingdom of Poland)", "FromYear": 1305, "ToYear": 1313, "Area": 313720.9193600344, "Type": "POLITY", "Wikipedia": "Wenceslaus II of Bohemia", "SeshatID": "cz_bohemian_k_1;pl_piast_dyn_2", "MemberOf": "", "Components": "Duchy of Jawor;Duchy of Głogów;(Kingdom of Poland);(Kingdom of Bohemia);Kingdom of Poland;Kingdom of Bohemia;Duchy of Legnica" },

So the personal union is its own composite with two seshat ids: cz_bohemian_k_1;pl_piast_dyn_2

Such was the nature of central european dynastic politics :) and the challenge to adapt that to seshat polities with clear political boundaries, thats why all this complication. Not sure how difficult this is and it only mainly concerns the HRE polities and components. Would still be nice if we could somehow reflect that

**Comment by edwardchalstrey1** [comment]
@Zsajk Here's how this example looks now in the year 1313 (note if you are reading this on 7th Jan Majid may be yet to update the website). Do you think anything is incorrect or not ideal in the way we display this now?

### World map

On the world map, we can see the personal union of bohemia and poland in 1313 looks like this - when it is selected, it includes the link to both Seshat pages in the text box in the top left

![Screenshot 2025-01-07 at 11 07 03](https://github.com/user-attachments/assets/6b18b806-3b42-4243-b3c5-75afa996cee6)

### Polish kingdom (809)

![Screenshot 2025-01-07 at 11 07 19](https://github.com/user-attachments/assets/97be9802-c06b-477e-b080-759af55e83a9)

### Bohemia (598)

![Screenshot 2025-01-07 at 11 07 29](https://github.com/user-attachments/assets/43c3dc3e-4238-4aca-973e-1d07e7b0db2c)

**Comment by Zsajk** [comment]
That looks very good, curious to try it out especially the single polity view . The world map view looks great. Is it already online at seshatdata.com ?

**Comment by edwardchalstrey1** [comment]
> That looks very good, curious to try it out especially the single polity view . The world map view looks great. Is it already online at seshatdata.com ?

Yes and on seshat-db as well now: https://seshat-db.com/core/polity/598

@Zsajk yes have a play and see if you notice any further inconsistencies

**Comment by Zsajk** [comment]
Ok, one issue I see with this new implantation of only showing top level polities on the worldmap:

For example for the year 1408:

![image](https://github.com/user-attachments/assets/3f0963b2-ba6b-4873-939a-19220d4f94bb)


We only see the HRE as one big polity but for the same year cz_bohemian_k_2 Kingdom of Bohemia also exists in the database as a polity: 

![image](https://github.com/user-attachments/assets/25c4307d-c4c9-4ad4-9340-06f32d9d752f)


They are connected via the [Suprapolity Relations](https://seshat-db.com/core/polity/568#Polity_suprapolity_relations) variable, Bohemia was a vassal to the HRE for that year. 

At the same time The Kingdom of Bohemia does not exist in the single polity view for the HRE for that year: 

![image](https://github.com/user-attachments/assets/1f5f361c-786b-40ac-b90d-65a5a9279ab4)

**Comment by edwardchalstrey1** [comment]
Ah ok, interesting! I guess the Supra Polity relations in Seshat isn't something I've explored yet. I can see if I can use this to show Cliopatria shapes that aren't part of the polity named in the polity page, but are from another polity related via personal union or vassalage. I can experiment with toggling on/off these relations and see what works best, or for example it might work to show vassals but not personal unions. I'll have a play around!

**Comment by Zsajk** [comment]
That would be great if this would allow to reflect changes on the maps directly by coding the suprapolity relations variable, this would be ideal I think .
